### PR TITLE
Add typescript/ts/tsx extension as a playground languages

### PR DESCRIPTION
**Pull request by mitsuruog**
Resolve:

- Code example does not render when the code block has `tsx padded` #1542

I added `typescript`, `ts`, `tsx` as a playground language.

It might make breaking current behavior. 
it's up to you whichever it goes to the current version or next major version.

### CODE REVIEW

File: src/loaders/utils/__tests__/chunkify.spec.ts [hunk 1]
```diff
@@ -45,6 +45,39 @@ This should be highlighted too:
...
+<h3>Hello playground!</h3>
+\`\`\`
+
+\`\`\`tsx
+<h3>Hello playground!</h3>
+\`\`\`
+`;
+
+	const actual = chunkify(markdown);
+	expect(actual).toMatchSnapshot();
```
Comment (sapegin) [inline src/loaders/utils/__tests__/chunkify.spec.ts]: Let's avoid a snapshot here: it's hard to see if it's correct or not.

File: src/loaders/utils/__tests__/chunkify.spec.ts [hunk 2]
```diff
@@ -45,6 +45,39 @@ This should be highlighted too:
 	expect(actual).toMatchSnapshot();
 });
 
+it('should some custom extensions render as a playground', () => {
```
Comment (sapegin) [inline src/loaders/utils/__tests__/chunkify.spec.ts]: Maybe?

```suggestion
it('should render some extensions as a playground', () => {
```

File: src/loaders/utils/__tests__/chunkify.spec.ts (hunk 1 above)
Comment (mitsuruog) [reply src/loaders/utils/__tests__/chunkify.spec.ts]: so should I write the test something like this?

```
const expected = [
  // expected result here
];
const actual = chunkify(markdown);
expect(actual).toEqual(expected);
```

Comment (sapegin) [reply src/loaders/utils/__tests__/chunkify.spec.ts]: Here you can check all of the chunks, except the first one, have the `code` type — that's the intent of the test. With a snapshot, it's impossible to tell what was the intent and whether it's correct.

**Comment by mitsuruog** [comment]
@sapegin I updated my test code. please review me again.

File: src/loaders/utils/__tests__/__snapshots__/chunkify.spec.ts.snap [hunk 3]
```diff
@@ -146,4 +146,4 @@ This should be highlighted too:
     "type": "markdown",
   },
 ]
-`;
+`;
```
Comment (mitsuruog) [inline src/loaders/utils/__tests__/__snapshots__/chunkify.spec.ts.snap]: ah... should I revert it?

Comment (sapegin) [reply src/loaders/utils/__tests__/__snapshots__/chunkify.spec.ts.snap]: It's not merged yet, so just update the test ;-)

**Comment by mitsuruog** [comment]
@sapegin I updated my test code. please review me again.

File: src/loaders/utils/__tests__/chunkify.spec.ts [hunk 4]
```diff
@@ -50,32 +50,69 @@ it('should render some extensions as a playground', () => {
...
+		{
+			type: 'code',
+			content: '<h3>Hello tsx playground!</h3>',
+			settings: {},
+		},
+	];
+
 	const actual = chunkify(markdown);
-	expect(actual).toMatchSnapshot();
+	expect(actual).toEqual(expected);
```
Comment (sapegin) [inline src/loaders/utils/__tests__/chunkify.spec.ts]: Why not something like, so the test intent is actually clear?

```suggestion
	expect(actual.slice(1).every(chunk => chunk.type === 'code')).toBe(true)
```

Not it's exactly the same as it was with a snapshot.

Comment (mitsuruog) [reply src/loaders/utils/__tests__/chunkify.spec.ts]: I don't copy as is what the snapshot did. I thought the current tests also can test for not playground extensions.
but I would like to follow your suggestion.

Comment (sapegin) [reply src/loaders/utils/__tests__/chunkify.spec.ts]: I didn't say you've copied but essentially it does the same.

**Comment by mitsuruog** [comment]
@sapegin I updated my test code. please review me again.

**Comment by sapegin** [comment]
The idea is _not to have_ a snapshot. Please have a look at my suggestion in my previous comment.

**Comment by mitsuruog** [comment]
I was misunderstood what you said. but now it is clear. thank you.
Previous tests compared objects with each other. That didn't tell exactly which point was testing.
but now I can see that your suggestion clearly checks only the `code` part.

@sapegin thank you for many times review.

**Comment by sapegin** [comment]
Merged manually to 11 branch: 5975c7cc722bf057231ad14115e47600a1fd98b2

**Pull request closed without merging**